colossalai/zero/sharded_param/sharded_tensor.py

```diff
@@ -1,30 +1,33 @@
 import torch
 import torch.distributed as dist
 from typing import Optional
+from colossalai.zero.sharded_param.tensorful_state import StatefulTensor, TensorState
 
 
-class ShardedTensor(object):
+class ShardedTensor(StatefulTensor):
 
     def __init__(self, tensor: torch.Tensor, process_group: Optional[dist.ProcessGroup] = None) -> None:
         r"""
         A tensor sharded in multiple processes. Constructed from an existing torch.Tensor instance.
         """
-        self._payload = tensor
-        self.process_group = process_group
-        self.world_size = dist.get_world_size(self.process_group)
-        self.local_rank = dist.get_rank(self.process_group)
-        self._is_sharded = False
+        super().__init__(tensor)
+        self.trans_state(TensorState.HOLD)
 
         self._origin_shape = tensor.shape
         self._origin_numel = tensor.numel()
         self._origin_dtype = tensor.dtype
 
+        self.process_group = process_group
```

Comment:
Does `ShardedTensor` still need process group now?

Reply:
DONE! removed the following 3 lines of code.